src/web/mjs/connectors/CoroCoro.mjs

```diff
@@ -0,0 +1,105 @@
+        this.tags = ['manga', 'japanese'];
+        this.url = 'https://www.corocoro.jp';
+    }
+
+    /**
+     * @param {URL} uri
+     */
+    async _getMangaFromURI(uri) {
+        if (uri.pathname.startsWith('/title/')) {
+            const id = uri.pathname.substring('/title/'.length);
+            /** @type {HTMLHeadingElement[]} */
+            const data = await this.fetchDOM(uri, 'h1');
+            const title = data[0].innerText.trim();
+            return new Manga(this, id, title);
+        }
+        if (uri.pathname.startsWith('/chapter/')) {
+            const scripts = await this.fetchDOM(uri, 'script');
+            const script = scripts[scripts.length - 1].innerText;
+            /** @type {string} */
+            const json = eval(script.substring(22, script.length - 2));
+            /** @type {{chapterListSection: {titleName: string}, viewerSection: {titleID: number}}} */
+            const data = JSON.parse(json.substring(json.indexOf(':') + 1))[3]['children'][0][3];
+            const id = data.viewerSection.titleID.toString();
+            const title = data.chapterListSection.titleName;
+            return new Manga(this, id, title);
+        }
+    }
+
+    /** @type {Connector['_getMangas']} */
+    async _getMangas() {
+        /** @type {HTMLScriptElement[]} */
+        const scripts = await this.fetchDOM(this.url + '/rensai', 'script');
+        const script = scripts[scripts.length - 2].innerText;
+        /** @type {string} */
+        const json = eval(script.substring(22, script.length - 2));
+        /** @type {Object<string, {id: number, name: string}[]>} */
+        const data = JSON.parse(json.substring(json.indexOf(':') + 1))[3]['children'][3]['weekdays'];
+        return weekdays.flatMap(weekday => data[weekday].map(manga => new Manga(this, manga.id.toString(), manga.name)));
+    }
+
+    /** @type {Connector['_getChapters']} */
+    async _getChapters(manga) {
+        const scripts = await this.fetchDOM(this.url + '/title/' + manga.id, 'script');
+        const script = scripts[scripts.length - 3].innerText;
+        /** @type {string} */
+        const json = eval(script.substring(22, script.length - 2));
+        /** @type {Object<string, {id: number, title: string}[]>} */
+        const data = JSON.parse(json.substring(json.indexOf(':') + 1))[3]['children'][0][3]['children'][1][3]['section']['chapters'];
+        return ['earlyChapters', 'omittedMiddleChapters', 'latestChapters']
+            .flatMap(key =>
+                data[key].map(chapter => new Chapter(manga, chapter.id.toString(), chapter.title, ''))
+            );
+    }
+
+    /** @type {Connector['_getPages']} */
+    async _getPages(chapter) {
+        const scripts = await this.fetchDOM(this.url + '/chapter/' + chapter.id, 'script');
+        const script = scripts[scripts.length - 1].innerText;
+        /** @type {string} */
+        const json = eval(script.substring(22, script.length - 2));
```

Comment:
not fond of evaluating code from website directly inside our code, for security reasons.